## PROYECTO 2 ##

**TÍTULO**

Objetivo: 

Identificar regiones de México con mayor potencial de desarrollo económico y social, combinando demografía, nivel educativo y condiciones de vivienda, para optimizar inversiones, servicios o políticas públicas.

In [147]:
import pandas as pd

url1 = 'https://raw.githubusercontent.com/julio-bp/PROYECTO2/refs/heads/main/datos_de_educacion.csv'
df_educacion = pd.read_csv(url1)

In [148]:
url2 = 'https://raw.githubusercontent.com/julio-bp/PROYECTO2/refs/heads/main/datos_poblacion.csv'
df_poblacion = pd.read_csv(url2)

In [149]:
url3 = 'https://raw.githubusercontent.com/julio-bp/PROYECTO2/refs/heads/main/datos_vivienda.csv'
df_vivienda = pd.read_csv(url3)

**Funciones**

In [150]:
## análisis general

In [151]:
def informe(x):
    print("Información del DataFrame:")
    print(x.info())
    print("Descripción estadística:")
    print(x.describe())
    print("Valores nulos:")
    print(x.isnull().sum())
    print("Tipos de datos:")
    print(x.dtypes)
    print("Primeras 5 filas:")
    print(x.head())
    print("Últimas 5 filas:")
    print(x.tail())
    print("Forma del DataFrame:")
    print(x.shape)
    print("columnas")
    print(x.columns)

In [152]:
## análisis funciones categóricas

In [153]:
def anali(x):
    print(x.value_counts())
    print(x.unique())
    print(x.nunique())

In [154]:
## limpieza 

In [155]:
def limpieza(df, column):
    df[column] = df[column].str.strip().str.upper()
    return df

In [156]:
## duplicados y nulos

In [157]:
def datos(x):
    print("datos nulos")
    print(x.isnull().sum())
    print("datos totales")
    print(x.count())   
    print("porcentaje de nulos")
    print(x.isnull().sum() / x.shape[0]*100)
    print("datos duplicados")
    print(x.duplicated().sum())

## ANÁLISIS ##

In [158]:
informe(df_vivienda)

Información del DataFrame:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 33 entries, 0 to 32
Data columns (total 37 columns):
 #   Column                                                                             Non-Null Count  Dtype  
---  ------                                                                             --------------  -----  
 0   Unnamed: 0                                                                         33 non-null     int64  
 1   estados                                                                            33 non-null     object 
 2   disponibilidad de bienes: automovil o camioneta                                    33 non-null     float64
 3   disponibilidad de bienes: bicicleta como medio de transporte                       33 non-null     float64
 4   disponibilidad de bienes: lavadora                                                 33 non-null     float64
 5   disponibilidad de bienes: motocicleta o motoneta                                 

In [159]:
informe(df_poblacion)

Información del DataFrame:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 33 entries, 0 to 32
Columns: 128 entries, Unnamed: 0 to indice de envejecimiento
dtypes: float64(126), int64(1), object(1)
memory usage: 33.1+ KB
None
Descripción estadística:
       Unnamed: 0  densidad de poblacion  edad mediana  edad mediana hombres  \
count    33.00000              33.000000     33.000000             33.000000   
mean     16.00000             302.240228     29.000000             28.212121   
std       9.66954            1062.566832      1.713914              1.709488   
min       0.00000              10.803055     24.000000             24.000000   
25%       8.00000              43.959693     28.000000             27.000000   
50%      16.00000              64.271664     29.000000             28.000000   
75%      24.00000             148.060870     30.000000             29.000000   
max      32.00000            6163.320469     35.000000             34.000000   

       edad mediana mujere

In [160]:
informe(df_educacion)

Información del DataFrame:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 33 entries, 0 to 32
Data columns (total 22 columns):
 #   Column                                                                                    Non-Null Count  Dtype  
---  ------                                                                                    --------------  -----  
 0   Unnamed: 0                                                                                33 non-null     int64  
 1   estados                                                                                   33 non-null     object 
 2   grado promedio de escolaridad de la poblacion de 15 y mas anos                            33 non-null     float64
 3   poblacion de 6 a 14 anos que no especifico si sabe o no leer y escribir                   33 non-null     float64
 4   poblacion de 6 a 14 anos que no sabe leer y escribir                                      33 non-null     float64
 5   poblacion de 6 a 14 anos que sab





### Importante para coordinar entre nosotros!!
Nombres de la Base de Datos:  mexico_stats
Ambos usamos .str.upper().str.strip(), para tener los mismos nombres de columnas!







**HIPÓTESIS 1**

Los estados con mayor nivel educativo presentan mejores condiciones generales de vivienda.

Fuentes: Educación y Vivienda

In [161]:
limpieza(df_educacion, "estados")

Unnamed: 0                          estados  \
0            0                   AGUASCALIENTES   
1            1                  BAJA CALIFORNIA   
2            2              BAJA CALIFORNIA SUR   
3            3                         CAMPECHE   
4            4                          CHIAPAS   
5            5                        CHIHUAHUA   
6            6                 CIUDAD DE MEXICO   
7            7             COAHUILA DE ZARAGOZA   
8            8                           COLIMA   
9            9                          DURANGO   
10          10         ESTADOS UNIDOS MEXICANOS   
11          11                       GUANAJUATO   
12          12                         GUERRERO   
13          13                          HIDALGO   
14          14                          JALISCO   
15          15              MICHOACAN DE OCAMPO   
16          16                          MORELOS   
17          17                           MEXICO   
18          18                          NAYARIT   
19          19                       NUEVO LEON   
20          20                           OAXACA   
21          21                           PUEBLA   
22          22                        QUERETARO   
23          23                     QUINTANA ROO   
24          24                  SAN LUIS POTOSI   
25          25                          SINALOA   
26          26                           SONORA   
27          27                          TABASCO   
28          28                       TAMAULIPAS   
29          29                         TLAXCALA   
30          30  VERACRUZ DE IGNACIO DE LA LLAVE   
31          31                          YUCATAN   
32          32                        ZACATECAS   

    grado promedio de escolaridad de la poblacion de 15 y mas anos  \
0                                           10.349134                
1                                           10.202655                
2                                           10.342548                
3                                            9.625130                
4                                            7.783218                
5                                            9.996942                
6                                           11.482039                
7                                           10.431996                
8                                           10.046730                
9                                            9.746069                
10                                           9.738050                
11                                           9.043787                
12                                           8.372699                
13                                           9.370759                
14                                           9.897353                
15                                           8.595490                
16                                           9.844782                
17                                          10.082398                
18                                           9.729524                
19                                          10.740959                
20                                           8.118536                
21                                           9.160472                
22                                          10.484599                
23                                          10.239289                
24                                           9.613127                
25                                          10.223508                
26                                          10.403197                
27                                           9.693396                
28                                          10.092209                
29                                           9.833170                
30                                           8.745629                
31                           

In [162]:
limpieza(df_vivienda, "estados")

Unnamed: 0                          estados  \
0            0                   AGUASCALIENTES   
1            1                  BAJA CALIFORNIA   
2            2              BAJA CALIFORNIA SUR   
3            3                         CAMPECHE   
4            4                          CHIAPAS   
5            5                        CHIHUAHUA   
6            6                 CIUDAD DE MEXICO   
7            7             COAHUILA DE ZARAGOZA   
8            8                           COLIMA   
9            9                          DURANGO   
10          10         ESTADOS UNIDOS MEXICANOS   
11          11                       GUANAJUATO   
12          12                         GUERRERO   
13          13                          HIDALGO   
14          14                          JALISCO   
15          15              MICHOACAN DE OCAMPO   
16          16                          MORELOS   
17          17                           MEXICO   
18          18                          NAYARIT   
19          19                       NUEVO LEON   
20          20                           OAXACA   
21          21                           PUEBLA   
22          22                        QUERETARO   
23          23                     QUINTANA ROO   
24          24                  SAN LUIS POTOSI   
25          25                          SINALOA   
26          26                           SONORA   
27          27                          TABASCO   
28          28                       TAMAULIPAS   
29          29                         TLAXCALA   
30          30  VERACRUZ DE IGNACIO DE LA LLAVE   
31          31                          YUCATAN   
32          32                        ZACATECAS   

    disponibilidad de bienes: automovil o camioneta  \
0                                         62.273096   
1                                         69.930111   
2                                         67.394865   
3                                         34.494526   
4                                         21.122835   
5                                         68.427459   
6                                         46.848904   
7                                         61.583405   
8                                         52.045309   
9                                         55.472663   
10                                        46.479608   
11                                        50.699758   
12                                        24.958696   
13                                        44.283327   
14                                        56.016297   
15                                        48.191124   
16                                        42.002533   
17                                        42.607359   
18                                        45.606936   
19                                        58.499008   
20                                        24.850202   
21                                        35.441912   
22                                        58.797986   
23                                        38.378120   
24                                        50.029415   
25                                        57.348835   
26                                        64.408966   
27                                        27.983533   
28                                        58.769281   
29                                        39.400330   
30                                        28.947727   
31                                        38.456737   
32                                        60.432593   

    disponibilidad de bienes: bicicleta como medio de transporte  \
0                                           27.396110              
1                                           13.559700              
2                                           11.886797              
3                                           23.887004              
4                                           13.912566     

In [163]:
df_vivienda.columns

Index(['Unnamed: 0', 'estados',
       'disponibilidad de bienes: automovil o camioneta',
       'disponibilidad de bienes: bicicleta como medio de transporte',
       'disponibilidad de bienes: lavadora',
       'disponibilidad de bienes: motocicleta o motoneta',
       'disponibilidad de bienes: refrigerador',
       'disponibilidad de servicios en la vivienda: cisterna o aljibe',
       'disponibilidad de servicios en la vivienda: tinaco',
       'porcentaje de viviendas con agua entubada dentro de la vivienda',
       'porcentaje de viviendas con electricidad',
       'porcentaje de viviendas con piso de tierra',
       'porcentaje de viviendas con techos de materiales resistentes',
       'porcentaje de viviendas particulares habitadas con disponibilidad de drenaje',
       'porcentaje de viviendas particulares habitadas con disponibilidad de excusado',
       'porcentaje de viviendas particulares habitadas propias',
       'porcentaje de viviendas particulares habitadas que dispo

In [164]:
df_h1_1 = df_educacion[["estados", "porcentaje de la poblacion de 15 anos y mas con instruccion media superior"]]

In [165]:
df_h1_1

estados  \
0                    AGUASCALIENTES   
1                   BAJA CALIFORNIA   
2               BAJA CALIFORNIA SUR   
3                          CAMPECHE   
4                           CHIAPAS   
5                         CHIHUAHUA   
6                  CIUDAD DE MEXICO   
7              COAHUILA DE ZARAGOZA   
8                            COLIMA   
9                           DURANGO   
10         ESTADOS UNIDOS MEXICANOS   
11                       GUANAJUATO   
12                         GUERRERO   
13                          HIDALGO   
14                          JALISCO   
15              MICHOACAN DE OCAMPO   
16                          MORELOS   
17                           MEXICO   
18                          NAYARIT   
19                       NUEVO LEON   
20                           OAXACA   
21                           PUEBLA   
22                        QUERETARO   
23                     QUINTANA ROO   
24                  SAN LUIS POTOSI   
25                          SINALOA   
26                           SONORA   
27                          TABASCO   
28                       TAMAULIPAS   
29                         TLAXCALA   
30  VERACRUZ DE IGNACIO DE LA LLAVE   
31                          YUCATAN   
32                        ZACATECAS   

    porcentaje de la poblacion de 15 anos y mas con instruccion media superior  
0                                           23.597657                           
1                                           27.979655                           
2                                           29.164156                           
3                                           22.269533                           
4                                           18.241452                           
5                                           23.805843                           
6                                           27.639769                           
7                                           24.913570                           
8                                           23.966872                           
9                                           22.676389                           
10                                          23.957897                           
11                                          21.025966                           
12                                          21.872415                           
13                                          23.003107                           
14                                          23.714334                           
15                                          19.171875                           
16                                          24.698057                           
17                                          27.125567                           
18                                          24.163447                           
19                                          25.475716                           
20                                          18.538463                           
21                                          21.865941                           
22                                          23.149962                           
23                                          28.733157                           
24                                          22.327073                           
25                                          24.987027                           
26                                          26.747444                           
27                                          25.433430                           
28                                          25.775652                           
29                                          24.484261                           
30                                          21.743559                           
31                                          22.552535                           
32                          

In [166]:
df_vivienda["indice_vivienda"] = (
    df_vivienda[
        [
            "porcentaje de viviendas con electricidad",
            "porcentaje de viviendas particulares habitadas con disponibilidad de drenaje",
            "porcentaje de viviendas particulares habitadas con disponibilidad de excusado"
        ]
    ].mean(axis=1)
)

In [167]:
df_h1_2 = df_vivienda[["estados", "porcentaje de viviendas con electricidad", "porcentaje de viviendas particulares habitadas con disponibilidad de excusado", "porcentaje de viviendas particulares habitadas con disponibilidad de drenaje", "indice_vivienda"]]

In [168]:
df_h1_2

estados  porcentaje de viviendas con electricidad  \
0                    AGUASCALIENTES                                 99.572551   
1                   BAJA CALIFORNIA                                 98.983702   
2               BAJA CALIFORNIA SUR                                 98.298365   
3                          CAMPECHE                                 98.167711   
4                           CHIAPAS                                 97.727773   
5                         CHIHUAHUA                                 98.291542   
6                  CIUDAD DE MEXICO                                 99.807534   
7              COAHUILA DE ZARAGOZA                                 99.516370   
8                            COLIMA                                 99.200247   
9                           DURANGO                                 97.977256   
10         ESTADOS UNIDOS MEXICANOS                                 99.001843   
11                       GUANAJUATO                                 99.303965   
12                         GUERRERO                                 97.968168   
13                          HIDALGO                                 98.992323   
14                          JALISCO                                 99.033892   
15              MICHOACAN DE OCAMPO                                 99.309950   
16                          MORELOS                                 99.518335   
17                           MEXICO                                 99.602686   
18                          NAYARIT                                 98.179288   
19                       NUEVO LEON                                 99.488756   
20                           OAXACA                                 97.499657   
21                           PUEBLA                                 99.130636   
22                        QUERETARO                                 99.150384   
23                     QUINTANA ROO                                 97.892616   
24                  SAN LUIS POTOSI                                 98.470311   
25                          SINALOA                                 99.363503   
26                           SONORA                                 98.921359   
27                          TABASCO                                 99.286746   
28                       TAMAULIPAS                                 99.196306   
29                         TLAXCALA                                 97.931686   
30  VERACRUZ DE IGNACIO DE LA LLAVE                                 98.606151   
31                          YUCATAN                                 98.916742   
32                        ZACATECAS                                 99.298607   

    porcentaje de viviendas particulares habitadas con disponibilidad de excusado  \
0                                           99.377479                               
1                                           99.340791                               
2                                           98.749572                               
3                                           96.383459                               
4                                           96.776290                               
5                                           98.464266                               
6                                           99.695949                               
7                                           99.291439                               
8                                           99.247058                               
9                                           96.920765                               
10                                          98.070381                               
11                                          97.440426                               
12                                          90.546592                               
13                                          97.262657                              

In [169]:
df_h1_3 = df_vivienda[["estados", "indice_vivienda"]]

In [170]:
df_h1_3

estados  indice_vivienda
0                    AGUASCALIENTES        99.402780
1                   BAJA CALIFORNIA        98.198239
2               BAJA CALIFORNIA SUR        97.972354
3                          CAMPECHE        96.318001
4                           CHIAPAS        95.170381
5                         CHIHUAHUA        97.755414
6                  CIUDAD DE MEXICO        99.739260
7              COAHUILA DE ZARAGOZA        99.110494
8                            COLIMA        99.226744
9                           DURANGO        96.541970
10         ESTADOS UNIDOS MEXICANOS        97.513852
11                       GUANAJUATO        97.786645
12                         GUERRERO        92.080829
13                          HIDALGO        97.044740
14                          JALISCO        98.780877
15              MICHOACAN DE OCAMPO        97.746414
16                          MORELOS        98.828538
17                           MEXICO        98.486014
18                          NAYARIT        96.994548
19                       NUEVO LEON        99.231854
20                           OAXACA        91.682578
21                           PUEBLA        97.503154
22                        QUERETARO        98.022157
23                     QUINTANA ROO        97.454383
24                  SAN LUIS POTOSI        95.728998
25                          SINALOA        98.115819
26                           SONORA        97.671999
27                          TABASCO        98.167052
28                       TAMAULIPAS        97.784539
29                         TLAXCALA        97.176941
30  VERACRUZ DE IGNACIO DE LA LLAVE        95.917924
31                          YUCATAN        94.878017
32                        ZACATECAS        97.550251

In [171]:
#tabla final y renombrada para SQL

df_h1 = df_h1_1.merge (
    df_h1_3,
    on = "estados",
    how = "inner"
)

In [172]:
df_h1

estados  \
0                    AGUASCALIENTES   
1                   BAJA CALIFORNIA   
2               BAJA CALIFORNIA SUR   
3                          CAMPECHE   
4                           CHIAPAS   
5                         CHIHUAHUA   
6                  CIUDAD DE MEXICO   
7              COAHUILA DE ZARAGOZA   
8                            COLIMA   
9                           DURANGO   
10         ESTADOS UNIDOS MEXICANOS   
11                       GUANAJUATO   
12                         GUERRERO   
13                          HIDALGO   
14                          JALISCO   
15              MICHOACAN DE OCAMPO   
16                          MORELOS   
17                           MEXICO   
18                          NAYARIT   
19                       NUEVO LEON   
20                           OAXACA   
21                           PUEBLA   
22                        QUERETARO   
23                     QUINTANA ROO   
24                  SAN LUIS POTOSI   
25                          SINALOA   
26                           SONORA   
27                          TABASCO   
28                       TAMAULIPAS   
29                         TLAXCALA   
30  VERACRUZ DE IGNACIO DE LA LLAVE   
31                          YUCATAN   
32                        ZACATECAS   

    porcentaje de la poblacion de 15 anos y mas con instruccion media superior  \
0                                           23.597657                            
1                                           27.979655                            
2                                           29.164156                            
3                                           22.269533                            
4                                           18.241452                            
5                                           23.805843                            
6                                           27.639769                            
7                                           24.913570                            
8                                           23.966872                            
9                                           22.676389                            
10                                          23.957897                            
11                                          21.025966                            
12                                          21.872415                            
13                                          23.003107                            
14                                          23.714334                            
15                                          19.171875                            
16                                          24.698057                            
17                                          27.125567                            
18                                          24.163447                            
19                                          25.475716                            
20                                          18.538463                            
21                                          21.865941                            
22                                          23.149962                            
23                                          28.733157                            
24                                          22.327073                            
25                                          24.987027                            
26                                          26.747444                            
27                                          25.433430                            
28                                          25.775652                            
29                                          24.484261                            
30                                          21.743559                            
31                                          22.552535                        

Comprobaciones y visualización:

**HIPÓTESIS 2**

Los estados con mayor presión demográfica presentan mayores niveles de hacinamiento en la vivienda.

In [173]:
limpieza(df_poblacion, "estados")

Unnamed: 0                          estados  densidad de poblacion  \
0            0                   AGUASCALIENTES             253.862309   
1            1                  BAJA CALIFORNIA              52.750459   
2            2              BAJA CALIFORNIA SUR              10.803055   
3            3                         CAMPECHE              16.149681   
4            4                          CHIAPAS              75.620723   
5            5                        CHIHUAHUA              15.124002   
6            6                 CIUDAD DE MEXICO            6163.320469   
7            7             COAHUILA DE ZARAGOZA              20.757782   
8            8                           COLIMA             129.981520   
9            9                          DURANGO              14.855625   
10          10         ESTADOS UNIDOS MEXICANOS              64.271664   
11          11                       GUANAJUATO             201.489845   
12          12                         GUERRERO              55.674754   
13          13                          HIDALGO             148.060870   
14          14                          JALISCO             106.216119   
15          15              MICHOACAN DE OCAMPO              81.040141   
16          16                          MORELOS             404.091463   
17          17                           MEXICO             760.227530   
18          18                          NAYARIT              44.350766   
19          19                       NUEVO LEON              90.161847   
20          20                           OAXACA              44.072675   
21          21                           PUEBLA             191.878340   
22          22                        QUERETARO             202.596169   
23          23                     QUINTANA ROO              41.560789   
24          24                  SAN LUIS POTOSI              46.162080   
25          25                          SINALOA              52.766039   
26          26                           SONORA              16.419082   
27          27                          TABASCO              97.149528   
28          28                       TAMAULIPAS              43.959693   
29          29                         TLAXCALA             336.027000   
30          30  VERACRUZ DE IGNACIO DE LA LLAVE             112.255486   
31          31                          YUCATAN              58.720617   
32          32                        ZACATECAS              21.549393   

    edad mediana  edad mediana hombres  edad mediana mujeres  \
0           27.0                  26.0                  28.0   
1           30.0                  30.0                  30.0   
2           29.0                  29.0                  30.0   
3           29.0                  29.0                  30.0   
4           24.0                  24.0                  25.0   
5           29.0                  29.0                  30.0   
6           35.0                  34.0                  36.0   
7           29.0                  28.0                  29.0   
8           30.0                  30.0                  31.0   
9           27.0                  27.0                  28.0   
10          29.0                  28.0                  30.0   
11          28.0                  27.0                  29.0   
12          27.0                  26.0                  28.0   
13          30.0                  28.0                  30.0   
14          29.0                  28.0                  30.0   
15          28.0                  27.0                  29.0   
16          30.0                  29.0                  32.0   
17          30.0                  29.0                  31.0   
18          29.0                  28.0                  30.0   
19          30.0                  29.0                  30.0   
20          28.0                  27.0                  29.0   
21          28.0                  26.0                  29.0   
22          29.

In [174]:
df_poblacion.rename(
    columns={"densidad de poblacion": "densidad_poblacion"},
    inplace=True
)

In [175]:
df_sel_poblacion = df_poblacion[["estados", "densidad_poblacion"]]
df_sel_poblacion

estados  densidad_poblacion
0                    AGUASCALIENTES          253.862309
1                   BAJA CALIFORNIA           52.750459
2               BAJA CALIFORNIA SUR           10.803055
3                          CAMPECHE           16.149681
4                           CHIAPAS           75.620723
5                         CHIHUAHUA           15.124002
6                  CIUDAD DE MEXICO         6163.320469
7              COAHUILA DE ZARAGOZA           20.757782
8                            COLIMA          129.981520
9                           DURANGO           14.855625
10         ESTADOS UNIDOS MEXICANOS           64.271664
11                       GUANAJUATO          201.489845
12                         GUERRERO           55.674754
13                          HIDALGO          148.060870
14                          JALISCO          106.216119
15              MICHOACAN DE OCAMPO           81.040141
16                          MORELOS          404.091463
17                           MEXICO          760.227530
18                          NAYARIT           44.350766
19                       NUEVO LEON           90.161847
20                           OAXACA           44.072675
21                           PUEBLA          191.878340
22                        QUERETARO          202.596169
23                     QUINTANA ROO           41.560789
24                  SAN LUIS POTOSI           46.162080
25                          SINALOA           52.766039
26                           SONORA           16.419082
27                          TABASCO           97.149528
28                       TAMAULIPAS           43.959693
29                         TLAXCALA          336.027000
30  VERACRUZ DE IGNACIO DE LA LLAVE          112.255486
31                          YUCATAN           58.720617
32                        ZACATECAS           21.549393

In [176]:
df_vivienda.rename(
    columns={"promedio de ocupantes por cuarto (indice de hacinamiento)": "indice_hacinamiento"},
    inplace=True
)

In [177]:
df_sel_vivienda = df_vivienda[["estados", "indice_hacinamiento"]]
df_sel_vivienda

estados  indice_hacinamiento
0                    AGUASCALIENTES             0.889100
1                   BAJA CALIFORNIA             0.898141
2               BAJA CALIFORNIA SUR             1.011096
3                          CAMPECHE             1.123364
4                           CHIAPAS             1.322805
5                         CHIHUAHUA             0.804774
6                  CIUDAD DE MEXICO             0.804542
7              COAHUILA DE ZARAGOZA             0.838291
8                            COLIMA             0.956519
9                           DURANGO             0.907989
10         ESTADOS UNIDOS MEXICANOS             0.958611
11                       GUANAJUATO             1.018931
12                         GUERRERO             1.294076
13                          HIDALGO             0.927177
14                          JALISCO             0.934264
15              MICHOACAN DE OCAMPO             1.016838
16                          MORELOS             0.956822
17                           MEXICO             0.948516
18                          NAYARIT             0.936082
19                       NUEVO LEON             0.849585
20                           OAXACA             1.138123
21                           PUEBLA             1.033391
22                        QUERETARO             0.911721
23                     QUINTANA ROO             1.137585
24                  SAN LUIS POTOSI             0.926317
25                          SINALOA             0.970587
26                           SONORA             0.884217
27                          TABASCO             1.098303
28                       TAMAULIPAS             0.910469
29                         TLAXCALA             0.996095
30  VERACRUZ DE IGNACIO DE LA LLAVE             0.977280
31                          YUCATAN             1.051168
32                        ZACATECAS             0.907704

In [178]:
# tabla final y renombrada para SQL

df_h2 = df_sel_poblacion.merge(
    df_sel_vivienda,
    on="estados",
    how="inner"
)

In [179]:
df_h2

estados  densidad_poblacion  indice_hacinamiento
0                    AGUASCALIENTES          253.862309             0.889100
1                   BAJA CALIFORNIA           52.750459             0.898141
2               BAJA CALIFORNIA SUR           10.803055             1.011096
3                          CAMPECHE           16.149681             1.123364
4                           CHIAPAS           75.620723             1.322805
5                         CHIHUAHUA           15.124002             0.804774
6                  CIUDAD DE MEXICO         6163.320469             0.804542
7              COAHUILA DE ZARAGOZA           20.757782             0.838291
8                            COLIMA          129.981520             0.956519
9                           DURANGO           14.855625             0.907989
10         ESTADOS UNIDOS MEXICANOS           64.271664             0.958611
11                       GUANAJUATO          201.489845             1.018931
12                         GUERRERO           55.674754             1.294076
13                          HIDALGO          148.060870             0.927177
14                          JALISCO          106.216119             0.934264
15              MICHOACAN DE OCAMPO           81.040141             1.016838
16                          MORELOS          404.091463             0.956822
17                           MEXICO          760.227530             0.948516
18                          NAYARIT           44.350766             0.936082
19                       NUEVO LEON           90.161847             0.849585
20                           OAXACA           44.072675             1.138123
21                           PUEBLA          191.878340             1.033391
22                        QUERETARO          202.596169             0.911721
23                     QUINTANA ROO           41.560789             1.137585
24                  SAN LUIS POTOSI           46.162080             0.926317
25                          SINALOA           52.766039             0.970587
26                           SONORA           16.419082             0.884217
27                          TABASCO           97.149528             1.098303
28                       TAMAULIPAS           43.959693             0.910469
29                         TLAXCALA          336.027000             0.996095
30  VERACRUZ DE IGNACIO DE LA LLAVE          112.255486             0.977280
31                          YUCATAN           58.720617             1.051168
32                        ZACATECAS           21.549393             0.907704

Comprobaciones y visualización:

# --- CARGA A MYSQL ---

In [180]:
# --- CARGA A MYSQL ---
engine = create_engine('mysql+pymysql://root:30802043@localhost:3306/mexico_stats')
df_h3.to_sql('hipotesis_3', con=engine, if_exists='replace', index=False)

33

# 📊 Hipótesis 1 (Bienestar - Julio B.)
        Objetivo: Ver si a mayor educación, mejores servicios (luz, agua) en la vivienda.

In [181]:
try:
    # 1. Seleccionar columnas usando los nombres exactos que nos dio el error
    col_edu = 'grado promedio de escolaridad de la poblacion de 15 y mas anos'
    col_agua = 'porcentaje de viviendas con agua entubada dentro de la vivienda'
    col_luz = 'viviendas particulares habitadas que disponen de energia electrica'

    # 2. Construir el DataFrame
    df_h1 = df_educacion[['estados', col_edu]].copy()
    df_h1['agua_entubada'] = df_vivienda[col_agua]
    df_h1['energia_electrica'] = df_vivienda[col_luz]
    
    # 3. Limpieza y Renombrado
    df_h1.columns = ['estados', 'escolaridad_promedio', 'agua_entubada', 'energia_electrica']
    df_h1['estados'] = df_h1['estados'].str.strip().str.upper()

    # 4. Carga a MySQL
    engine = create_engine('mysql+pymysql://root:30802043@localhost:3306/mexico_stats')
    df_h1.to_sql('hipotesis_1', con=engine, if_exists='replace', index=False)
    
    print("✅ Hipótesis 1 (Julio) cargada con éxito.")
    display(df_h1.head(3))

except Exception as e:
    print(f"❌ Error en Hipótesis 1: {e}")

✅ Hipótesis 1 (Julio) cargada con éxito.


estados  escolaridad_promedio  agua_entubada  energia_electrica
0       AGUASCALIENTES             10.349134      95.267751             384361
1      BAJA CALIFORNIA             10.202655      92.908462            1132622
2  BAJA CALIFORNIA SUR             10.342548      82.290962             235285

# 📊 Etapa 2: Hipótesis 2 (Presión Demográfica - Julio B.)
     Objetivo: Ver si los estados con más gente tienen más hacinamiento.

In [182]:
try:
    # 1. Seleccionar columnas
    col_pob = 'poblacion total'
    # Usamos la columna de ocupantes que aparece en tu lista
    col_hac = 'promedio de ocupantes en viviendas particulares habitadas'

    df_h2 = df_poblacion[['estados', col_pob]].copy()
    df_h2['indice_hacinamiento'] = df_vivienda[col_hac]
    
    # 2. Limpieza y Renombrado
    df_h2.columns = ['estados', 'pob_total', 'indice_hacinamiento']
    df_h2['estados'] = df_h2['estados'].str.strip().str.upper()

    # 3. Carga a MySQL
    engine = create_engine('mysql+pymysql://root:30802043@localhost:3306/mexico_stats')
    df_h2.to_sql('hipotesis_2', con=engine, if_exists='replace', index=False)
    
    print("✅ Hipótesis 2 (Julio) cargada con éxito.")
    display(df_h2.head(3))

except Exception as e:
    print(f"❌ Error en Hipótesis 2: {e}")

✅ Hipótesis 2 (Julio) cargada con éxito.


estados  pob_total  indice_hacinamiento
0       AGUASCALIENTES  1425607.0             3.678621
1      BAJA CALIFORNIA  3769020.0             3.258335
2  BAJA CALIFORNIA SUR   798447.0             3.303420

**HIPÓTESIS 3**

Los estados con mayor nivel educativo concentran una mayor proporción de población en edad productiva y menor envejecimiento.

In [183]:
df_educacion[['estados', 'grado promedio de escolaridad de la poblacion de 15 y mas anos']]

estados  \
0                    AGUASCALIENTES   
1                   BAJA CALIFORNIA   
2               BAJA CALIFORNIA SUR   
3                          CAMPECHE   
4                           CHIAPAS   
5                         CHIHUAHUA   
6                  CIUDAD DE MEXICO   
7              COAHUILA DE ZARAGOZA   
8                            COLIMA   
9                           DURANGO   
10         ESTADOS UNIDOS MEXICANOS   
11                       GUANAJUATO   
12                         GUERRERO   
13                          HIDALGO   
14                          JALISCO   
15              MICHOACAN DE OCAMPO   
16                          MORELOS   
17                           MEXICO   
18                          NAYARIT   
19                       NUEVO LEON   
20                           OAXACA   
21                           PUEBLA   
22                        QUERETARO   
23                     QUINTANA ROO   
24                  SAN LUIS POTOSI   
25                          SINALOA   
26                           SONORA   
27                          TABASCO   
28                       TAMAULIPAS   
29                         TLAXCALA   
30  VERACRUZ DE IGNACIO DE LA LLAVE   
31                          YUCATAN   
32                        ZACATECAS   

    grado promedio de escolaridad de la poblacion de 15 y mas anos  
0                                           10.349134               
1                                           10.202655               
2                                           10.342548               
3                                            9.625130               
4                                            7.783218               
5                                            9.996942               
6                                           11.482039               
7                                           10.431996               
8                                           10.046730               
9                                            9.746069               
10                                           9.738050               
11                                           9.043787               
12                                           8.372699               
13                                           9.370759               
14                                           9.897353               
15                                           8.595490               
16                                           9.844782               
17                                          10.082398               
18                                           9.729524               
19                                          10.740959               
20                                           8.118536               
21                                           9.160472               
22                                          10.484599               
23                                          10.239289               
24                                           9.613127               
25                                          10.223508               
26                                          10.403197               
27                                           9.693396               
28                                          10.092209               
29                                           9.833170               
30                                           8.745629               
31                                           9.594718               
32                                           9.245563

In [184]:
df_poblacion[['estados', 'indice de envejecimiento','poblacion total']]

estados  indice de envejecimiento  poblacion total
0                    AGUASCALIENTES                 37.740884        1425607.0
1                   BAJA CALIFORNIA                 43.185780        3769020.0
2               BAJA CALIFORNIA SUR                 38.369373         798447.0
3                          CAMPECHE                 42.558415         928363.0
4                           CHIAPAS                 28.685604        5543828.0
5                         CHIHUAHUA                 44.551709        3741869.0
6                  CIUDAD DE MEXICO                 90.249478        9209944.0
7              COAHUILA DE ZARAGOZA                 42.283811        3146771.0
8                            COLIMA                 52.873304         731391.0
9                           DURANGO                 40.762088        1832650.0
10         ESTADOS UNIDOS MEXICANOS                 47.686476      126014024.0
11                       GUANAJUATO                 41.656620        6166934.0
12                         GUERRERO                 42.855141        3540685.0
13                          HIDALGO                 48.479438        3082841.0
14                          JALISCO                 47.485277        8348151.0
15              MICHOACAN DE OCAMPO                 46.344287        4748846.0
16                          MORELOS                 58.457457        1971520.0
17                           MEXICO                 46.565328       16992418.0
18                          NAYARIT                 47.568892        1235456.0
19                       NUEVO LEON                 47.345249        5784442.0
20                           OAXACA                 48.002570        4132148.0
21                           PUEBLA                 41.860916        6583278.0
22                        QUERETARO                 41.171762        2368467.0
23                     QUINTANA ROO                 28.651509        1857985.0
24                  SAN LUIS POTOSI                 49.776994        2822255.0
25                          SINALOA                 52.686691        3026943.0
26                           SONORA                 49.513778        2944840.0
27                          TABASCO                 39.399021        2402598.0
28                       TAMAULIPAS                 48.905210        3527735.0
29                         TLAXCALA                 41.931386        1342977.0
30  VERACRUZ DE IGNACIO DE LA LLAVE                 59.048918        8062579.0
31                          YUCATAN                 52.433129        2320898.0
32                        ZACATECAS                 43.543430        1622138.0

In [185]:
from sqlalchemy import create_engine

# --- TRANSFORMACIÓN HIPÓTESIS 3 ---
try:
    # Seleccionamos columnas de Educación y Población
    df_h3 = df_educacion[['estados', 'grado promedio de escolaridad de la poblacion de 15 y mas anos']].copy()
    
    # Añadimos datos de la tabla de población
    df_h3['indice_envejecimiento'] = df_poblacion['indice de envejecimiento']
    df_h3['pob_total'] = df_poblacion['poblacion total']
    
    # Renombramos para que sea fácil en SQL
    df_h3.columns = ['estados', 'escolaridad_promedio', 'indice_envejecimiento', 'pob_total']
    
    # Limpieza: Estados en mayúsculas
    df_h3['estados'] = df_h3['estados'].str.strip().str.upper()


except Exception as e:
    print(f"❌ Error en Hipótesis 3: {e}")

Comprobaciones y visualización:

In [186]:
print("✅ Hipótesis 3: Cargada con éxito en MySQL.")
display(df_h3.head(5))

✅ Hipótesis 3: Cargada con éxito en MySQL.


estados  escolaridad_promedio  indice_envejecimiento  pob_total
0       AGUASCALIENTES             10.349134              37.740884  1425607.0
1      BAJA CALIFORNIA             10.202655              43.185780  3769020.0
2  BAJA CALIFORNIA SUR             10.342548              38.369373   798447.0
3             CAMPECHE              9.625130              42.558415   928363.0
4              CHIAPAS              7.783218              28.685604  5543828.0

**HIPÓTESIS 4**

Los estados con mayor nivel educativo presentan un mayor acceso a internet en las viviendas.

In [187]:
# --- TRANSFORMACIÓN HIPÓTESIS 4 ---
try:
    # Seleccionamos columnas de Educación y Vivienda
    df_h4 = df_educacion[['estados', 'grado promedio de escolaridad de la poblacion de 15 y mas anos']].copy()
    
    # Añadimos datos de la tabla de vivienda (Internet y Computadoras)
    df_h4['porcentaje_internet'] = df_vivienda['porcentaje de viviendas particulares habitadas que disponen de internet']
    df_h4['viviendas_con_computadora'] = df_vivienda['viviendas particulares habitadas que disponen de computadora']
    
    # Renombramos
    df_h4.columns = ['estados', 'escolaridad_promedio', 'porcentaje_internet', 'viviendas_con_computadora']
    
    # Limpieza: Estados en mayúsculas
    df_h4['estados'] = df_h4['estados'].str.strip().str.upper() 

except Exception as e:
    print(f"❌ Error en Hipótesis 4: {e}")

Comprobaciones y visualización:

In [188]:
print("✅ Hipótesis 4: Cargada con éxito en MySQL.")
display(df_h4.head(5))

✅ Hipótesis 4: Cargada con éxito en MySQL.


estados  escolaridad_promedio  porcentaje_internet  \
0       AGUASCALIENTES             10.349134            61.138931   
1      BAJA CALIFORNIA             10.202655            69.931248   
2  BAJA CALIFORNIA SUR             10.342548            62.134125   
3             CAMPECHE              9.625130            43.816602   
4              CHIAPAS              7.783218            21.674054   

   viviendas_con_computadora  
0                     177149  
1                     576454  
2                     108948  
3                      87769  
4                     212970

In [189]:
# --- CARGA A MYSQL ---
engine = create_engine('mysql+pymysql://root:30802043@localhost:3306/mexico_stats')
df_h3.to_sql('hipotesis_3', con=engine, if_exists='replace', index=False)
    

33

#  📊 Hipótesis 3: Educación vs. Envejecimiento
    ¿Los estados más educados tienen poblaciones más envejecidas?

In [190]:
# 1. Consulta para Hipótesis 3
query_h3 = """
SELECT estados, escolaridad_promedio, indice_envejecimiento 
FROM hipotesis_3 
ORDER BY escolaridad_promedio DESC 
LIMIT 10;
"""

try:
    df_analisis_h3 = pd.read_sql(query_h3, con=engine)
    
    print("👵 Top 10 Estados con mayor Escolaridad y su Índice de Envejecimiento:")
    display(df_analisis_h3)

    # 2. Cálculo de correlación
    correlacion_h3 = df_analisis_h3['escolaridad_promedio'].corr(df_analisis_h3['indice_envejecimiento'])
    print(f"\n💡 El grado de relación entre educación y envejecimiento es de: {correlacion_h3:.2f}")
    
except Exception as e:
    print(f"❌ Error al consultar SQL: {e}")

👵 Top 10 Estados con mayor Escolaridad y su Índice de Envejecimiento:


estados  escolaridad_promedio  indice_envejecimiento
0      CIUDAD DE MEXICO             11.482039              90.249478
1            NUEVO LEON             10.740959              47.345249
2             QUERETARO             10.484599              41.171762
3  COAHUILA DE ZARAGOZA             10.431996              42.283811
4                SONORA             10.403197              49.513778
5        AGUASCALIENTES             10.349134              37.740884
6   BAJA CALIFORNIA SUR             10.342548              38.369373
7          QUINTANA ROO             10.239289              28.651509
8               SINALOA             10.223508              52.686691
9       BAJA CALIFORNIA             10.202655              43.185780


💡 El grado de relación entre educación y envejecimiento es de: 0.87


Tendencia Demográfica: La correlación es positiva, confirma que los estados con mayor acceso a la educación tienden a tener una transición demográfica más avanzada. Esto suele ocurrir porque a mayor escolaridad, se posterga la maternidad/paternidad y mejora la esperanza de vida.

Contraste Geográfico: Vemos que la CDMX probablemente lidera en ambos indicadores, mientras que otros estados del norte tienen mucha escolaridad pero poblaciones un poco más jóvenes (menos envejecidas) debido a la migración interna.

#  📊 Análisis de la Hipótesis 4: Educación vs. Internet
         1. Definir la consulta SQL (Query)

In [191]:
import pandas as pd

# Queremos ver los estados ordenados por nivel educativo de mayor a menor
query_h4 = """
SELECT estados, escolaridad_promedio, porcentaje_internet 
FROM hipotesis_4 
ORDER BY escolaridad_promedio DESC 
LIMIT 10;
"""

# 2. Traer los datos de MySQL a Python para analizar
try:
    df_analisis_h4 = pd.read_sql(query_h4, con=engine)
    
    print("📈 Top 10 Estados con mayor Escolaridad y su acceso a Internet:")
    display(df_analisis_h4)

    # 3. Un pequeño cálculo de correlación (Insight rápido)
    correlacion = df_analisis_h4['escolaridad_promedio'].corr(df_analisis_h4['porcentaje_internet'])
    print(f"\n💡 El grado de relación entre educación e internet es de: {correlacion:.2f}")
    
except Exception as e:
    print(f"❌ Error al consultar SQL: {e}")

📈 Top 10 Estados con mayor Escolaridad y su acceso a Internet:


estados  escolaridad_promedio  porcentaje_internet
0      CIUDAD DE MEXICO             11.482039            75.727762
1            NUEVO LEON             10.740959            69.575305
2             QUERETARO             10.484599            64.416953
3  COAHUILA DE ZARAGOZA             10.431996            57.715587
4                SONORA             10.403197            60.675526
5        AGUASCALIENTES             10.349134            61.138931
6   BAJA CALIFORNIA SUR             10.342548            62.134125
7          QUINTANA ROO             10.239289            55.596700
8               SINALOA             10.223508            51.681644
9       BAJA CALIFORNIA             10.202655            69.931248


💡 El grado de relación entre educación e internet es de: 0.72


Existe una relación clara donde, a medida que sube el nivel educativo de un estado, el acceso a internet también tiende a subir de forma significativa.

       La Ciudad de México encabeza ambas listas, lo cual tiene sentido por ser el centro económico y educativo.

El "Outlier" (Caso Curioso): Baja California. Aunque está en el lugar 10 de escolaridad (10.20), tiene casi el 70% de internet, superando a Nuevo León que tiene más escolaridad.

       Esto sugiere que la cercanía con la frontera o factores económicos también impulsan el internet, no solo la escuela.

# Graficas por hipotesis